### PR TITLE
Remove global promise override

### LINKED ISSUE #11943: [Tech Debt] Remove override of `global.Promise` to Bluebird
The global `Promise` constructor in Ghost is [overridden to be the Bluebird Promise](https://github.com/TryGhost/Ghost/blob/master/core/server/overrides.js#L27-L30
), since [issue #9064](https://github.com/TryGhost/Ghost/issues/9064)

Since we now use `async`/`await` throughout the codebase we can no longer rely on all instances of promises being actual bluebird promise's because async functions do not return `global.Promise` they just return "real" promises created by the JS engine. 

We should remove the override, and fix code using bluebird promise methods to explicitly require them in.

### PR DESCRIPTION
**Pull request by sainthkh**
Closes #11943 

Currently, there's a bug in `brute-knex` when we don't set global `Promise` to `bluebird`(llambda/brute-knex#11). Because of that, I had to use the fixed version of mine. 

I'm waiting for the PR(llambda/brute-knex#13) to be merged and released a new version.

----

- [x] There's a clear use-case for this code change -> We can use more `await` syntax.
- [x] Commit message has a short title & references relevant issues
- [x] The build will pass (run `yarn test` and `yarn lint`)

----

Debt paid in full.

### CODE REVIEW

**Review by allouis — COMMENTED** [review]
The changes here look great! Thanks for contributing :relaxed: 

Unfortunately we do not want to use a fork of `brute-knex` unless absolutely necessary - would you be able to chase up your PR there and ping us on this when it's merged?

Thanks!

**Comment by sainthkh** [comment]
It seems that the author of `brute-knex` library updates it every few months... 

I hope it is updated in 2020 😄 .

**Comment by daniellockyer** [comment]
@sainthkh I gave the maintainer a ping, and they instantly merged and released 4.0.1 :grinning: So you should be good to put it back to the official version and rebase your PR :slightly_smiling_face:

**Comment by sainthkh** [comment]
@daniellockyer Waiting for the merge.

**Review by allouis — APPROVED** [review]
Looks good to me, @daniellockyer do you mind approving too?

**Review by daniellockyer — APPROVED** [review]
LGTM!

**Comment by allouis** [comment]
Thanks @sainthkh :tada:

**Pull request merged**
